### PR TITLE
feat: modify  jx cloudbees to open the UI app in local, read-only mode

### PR DESCRIPTION
**Pull request by dgozalo**
feat: refactored the way we run this command to do port-forwarding

feat: find resources by label not by name to avoid problems with aliases or releases

fix: remove unusued code

feat: added tests and refactored a bit
Signed-off-by: Daniel Gozalo <dgozalo@cloudbees.com>

#### Submitter checklist

- [X] Change is code complete and matches issue description.
- [X] Change is covered by existing or new tests.

#### Description
Modification to open the UI in local, read-only mode with `kubectl port-forward` while ignoring, for now, the code to open it in SSO mode.

It runs the `kubectl port-forward` command quietly unless the `verbose` flag is passed, it also does it in a goroutine to avoid blocking the main routine which will open the UI once the ready check is passed.

Lastly, I removed `cloudbees_pipeline.go` because it was still pointing to the old UI paths.

### CODE REVIEW

File: pkg/cmd/cloudbees/cloudbees.go [hunk 1]
```diff
@@ -56,49 +66,147 @@ func NewCmdCloudBees(commonOpts *opts.CommonOptions) *cobra.Command {
...
+		cmd := fmt.Sprintf("kubectl port-forward service/%s %s:80", serviceName, o.LocalPort)
+		log.Logger().Debugf("Executed command: %s", util.ColorInfo(cmd))
+		outWriter = o.Out
+	}
+	c := exec.Command("kubectl", "port-forward", fmt.Sprintf("service/%s", serviceName), fmt.Sprintf("%s:80", o.LocalPort))
+	c.Stdout = outWriter
+	c.Stderr = o.Err
+	err := c.Run()
 	if err != nil {
-		return err
```
[the remark this answers is not in the document]
Comment (warrenbailey) [reply pkg/cmd/cloudbees/cloudbees.go]: cool

File: pkg/cmd/cloudbees/cloudbees.go [hunk 2]
```diff
@@ -56,49 +66,147 @@ func NewCmdCloudBees(commonOpts *opts.CommonOptions) *cobra.Command {
...
+	if err != nil {
+		return "", "", err
+	}
+	kubeClient, err := o.KubeClient()
+	if err != nil {
+		return "", "", err
+	}
+	apps, err := jxClient.JenkinsV1().Apps(ns).List(listOptions)
+	if err != nil || len(apps.Items) == 0 {
+		log.Logger().Errorf("Couldn't find the jxui app installed in the cluster. Did you add it via %s?", util.ColorInfo("jx add app ui"))
```
[the remark this answers is not in the document]
Comment (warrenbailey) [reply pkg/cmd/cloudbees/cloudbees.go]: ah cool, didn't know that

File: pkg/cmd/cloudbees/cloudbees.go [hunk 3]
```diff
@@ -56,49 +67,149 @@ func NewCmdCloudBees(commonOpts *opts.CommonOptions) *cobra.Command {
...
+		return "", "", errors.Wrap(err, "there was an error obtaining the local port to forward to")
+	}
+
+	serviceName := services.Items[0].Name
+	log.Logger().Debugf("Found UI service name %s", util.ColorInfo(serviceName))
+
+	return fmt.Sprintf("http://localhost:%s", o.LocalPort), serviceName, nil
+}
+
+func (o CloudBeesOptions) waitForForwarding(localURL string) error {
```
Comment (hferentschik) [inline pkg/cmd/cloudbees/cloudbees.go]: `waitForURL`?

**Pull request merged**